msbuild/Xamarin.MacDev.Tasks.Core/VerbosityUtils.cs

```diff
@@ -41,50 +41,58 @@ static public string[] Merge (string extraArguments, LoggerVerbosity taskVerbosi
 		//
 		static public string[] GetVerbosityLevel (string commandLine)
 		{
-			const string shortForm = "v:";
-			const string longForm = "verbosity:";
-
 			if (!StringUtils.TryParseArguments (commandLine, out var args, out _))
 				return GetVerbosityLevel (LoggerVerbosity.Normal);
 
+			var hasBinaryLog = false;
 			foreach (var arg in args) {
-				// the minimum length we're looking for is `/v:q`
-				if (arg.Length < 4)
+				// the minimum length we're looking for is `/bl`
+				if (arg.Length < 2)
 					continue;
 				// msbuild accepts two types of argument separator
 				if (arg [0] != '/' && arg [0] != '-')
 					continue;
 
-				var verbosity = String.Empty;
-				// short and long forms are possible, always case sensitive
-				if (arg.IndexOf (shortForm, StringComparison.Ordinal) == 1) {
-					// skip argument prefix (+1) and the (short) argument to get the verbosity level
-					verbosity = arg.Substring (shortForm.Length + 1);
-				} else if (arg.IndexOf (longForm, StringComparison.Ordinal) == 1) {
-					verbosity = arg.Substring (longForm.Length + 1);
-				} else
-					continue;
 
-				// case sensitive
-				switch (verbosity) {
-				case "q":
-				case "quiet":
-					return GetVerbosityLevel (LoggerVerbosity.Quiet);
-				case "m":
-				case "minimal":
-					return GetVerbosityLevel (LoggerVerbosity.Minimal);
-				case "n":
-				case "normal":
-				default:
-					return GetVerbosityLevel (LoggerVerbosity.Normal);
-				case "d":
-				case "detailed":
-					return GetVerbosityLevel (LoggerVerbosity.Detailed);
-				case "diag":
-				case "diagnostic":
-					return GetVerbosityLevel (LoggerVerbosity.Diagnostic);
+				var colon = arg.IndexOf (':');
+				var name = arg.Substring (1, colon == -1 ? arg.Length - 1 : colon - 1);
+				var value = colon == -1 ? string.Empty : arg.Substring (colon + 1);
+
+				// the argument is not case sensitive
+				switch (name.ToLowerInvariant ()) {
+				case "v":
+				case "verbosity":
+					var verbosity = value;
+					// case sensitive
+					switch (verbosity) {
+					case "q":
+					case "quiet":
+						return GetVerbosityLevel (LoggerVerbosity.Quiet);
+					case "m":
+					case "minimal":
+						return GetVerbosityLevel (LoggerVerbosity.Minimal);
+					case "n":
+					case "normal":
+					default:
+						return GetVerbosityLevel (LoggerVerbosity.Normal);
+					case "d":
+					case "detailed":
+						return GetVerbosityLevel (LoggerVerbosity.Detailed);
+					case "diag":
+					case "diagnostic":
+						return GetVerbosityLevel (LoggerVerbosity.Diagnostic);
+					}
+				case "bl":
+				case "binarylogger":
+					hasBinaryLog = true;
```

Comment:
Unlike the verbosity options you're not returning when you first see `-bl`, so if you provide `-bl -verbosity:quiet` then you end up with `quiet`.

Not sure what would be the best interpretation but the commit message implies otherwise ("show verbose").

Reply:
I changed the commit message as you suggested.

The intention was to default to diagnostic mtouch/mmp unless `-verbosity:...` was passed, but I'm not sure what's the best behavior is, I figured we could just try this and see how it works out. Showing verbose output when asked to be quiet doesn't seem right, but that would also mean the binary log won't have the verbose output either.

Reply:
Parsing conflicting (or less logical) arguments is not easy. The early return here does not match how `mtouch` (or `mmp`) handle multiple arguments - normally the last, not the first, one wins. Still that's the current behaviour...